Add tests for `assert_follows_unicode_guidelines()`

eerovaher opened the pull request:
### Description

`assert_follows_unicode_guidelines()` is a function meant to help ensure our code follows our Unicode guidelines, but there aren't any test to ensure the function itself works as intended. I have written tests to trigger each `assert` in `assert_follows_unicode_guidelines()` and also a few tests to check that what is allowed by our Unicode guidelines is also allowed by the function.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

pllim commented:
Derek had reviewed the unicode stuff in the past, so I'll defer to his expertise. Thanks!

eerovaher committed:
Add tests for assert_follows_unicode_guidelines()

The function is a test helper meant to ensure code follows `astropy`
Unicode guidelines, but so far there have been no tests to ensure that
the test helper itself works as intended.
